src/components/Slider/SliderBase/SliderBase.jsx

```diff
@@ -0,0 +1,90 @@
+import React, { forwardRef } from "react";
+import PropTypes from "prop-types";
+import "./SliderBase.scss";
+import { bem } from "../SliderCommons";
+import { useSliderActions, useSliderSelection, useSliderUi } from "../SliderContext";
+import { calcDimensions, getNearest, moveToPx } from "../SliderHelpers";
+import { useSliderRail } from "../SliderHooks";
+import SliderRail from "./SliderRail";
+import SliderTrack from "./SliderTrack";
+import SliderFilledTrack from "./SliderFilledTrack";
+import SliderThumb from "./SliderThumb";
+import { isArrowDownEvent, isArrowLeftEvent, isArrowRightEvent, isArrowUpEvent } from "../../../utils/dom-event-utils";
+
+function getKey(index) {
+  return index;
+}
+
+const SliderBase = forwardRef(({ className }, ref) => {
+  const { color, disabled, dragging, size, consumerBem, shapeTestId } = useSliderUi();
+  const { isRange, min, max, step, value } = useSliderSelection();
+  const { changeValue, increaseValue, decreaseValue } = useSliderActions();
+  const { railCoords, railRef } = useSliderRail({ min, max, step, ref });
+  const { dimension, offset, positions } = calcDimensions({ isRange, max, min, value });
+
+  function handlePointerMove(e) {
```

Comment:
If I remember correctly, when the functions defined inside the functional component it will create new instance of the functions for each render.
Isn't it will be more efficient to use here useCallback here for define the functions?